src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,149 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        tail = new Node<>(null, null, null);
+        head = new Node<>(null, null, null);
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (size == 0) {
+            tail = head = node;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        Node<T> node = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = node;
+        } else if (index == 0) {
+            node.next = head;
+            head.prev = node;
+            tail = node;
+            head = node;
+            size++;
+        } else if (index == size) {
+            add(value);
+        } else {
+            Node<T> current = getNodeByIndex(index - 1);
+            node.next = current.next;
+            node.prev = current;
+            current.next = node;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkEqualsIndex(index);
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkEqualsIndex(index);
+        Node<T> node = getNodeByIndex(index);
+        T prevValue = node.value;
+        node.value = value;
+        return prevValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkEqualsIndex(index);
+        Node<T> node = getNodeByIndex(index);
+        unlink(node);
+        return node.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        while (current != null) {
+            if (object == current.value || current.value != null && current.value.equals(object)) {
+                unlink(current);
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Wrong index" + index);
+        }
+    }
```

Comment:
```suggestion
       
```
maybe better remove it and move all logic to add(T value, int index) {
         if (index < 0 || index > size) {
            throw new IndexOutOfBoundsException("Wrong index" + index);
        }
        
     
}